framework-generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/EventSourcesDestinationTypeFinder.java

```diff
@@ -0,0 +1,108 @@
+     * @param destinationName  the matching destination name
+     * @return Queue, Topic or empty
+     */
+    public Optional<Class<? extends Destination>> findForEventProcessor(final String serviceComponent, final String destinationName) {
+
+        if (serviceComponent == null || destinationName == null || !uriToDestinationTypeConverter.isEventProcessor(serviceComponent)) {
+            return empty();
+        }
+
+        try {
+            final EventSourcesFileParser eventSourcesFileParser = eventSourceYamlParserFactory.create();
+            final ArrayList<Path> list = new ArrayList<>();
+            final List<EventSourceDefinition> eventSourceDefinitionList = eventSourcesFileParser.getEventSourceDefinitions(Path.of(""), list);
+            if (eventSourceDefinitionList == null) {
+                LOGGER.debug("No SubscriptionWrapper found .. returning");
+                return empty();
+            }
+
+            LOGGER.debug("eventSourceDefinitionList size is {} \n eventSourceDefinitionList are {}\n destinationName is {}",
+                    eventSourceDefinitionList.size(), eventSourceDefinitionList, destinationName);
+
+            final List<String> foundUris = eventSourceDefinitionList.stream().map(eventSourceDefinition -> eventSourceDefinition.getLocation().getJmsUri()).
+                    filter(jmsUri -> destinationName.equalsIgnoreCase(uriToDestination(jmsUri))).toList();
+
+            LOGGER.debug("Found jsmUris are {}", foundUris);
+
+            if (foundUris.isEmpty()) {
+                LOGGER.debug("No filtered jmcUris found .. returning");
+                return empty();
+            }
+
+            final Set<Optional<Class<? extends Destination>>> destinations = foundUris.stream().
+                    map(uri -> uriToDestinationTypeConverter.convertForEventProcessor(serviceComponent, uri)).
+                    filter(destinationType -> destinationType.isPresent()).
+                    collect(Collectors.toSet());
+
+            if (destinations.isEmpty()) {
+                LOGGER.debug("No destination found ... returning");
+                return empty();
+            }
+
+            if (destinations.size() > 1) {
+                LOGGER.debug("Too many destination found ... returning");
+                return empty();
+            }
+
+            final Optional<Class<? extends Destination>> finalDestination = destinations.stream().
+                    findFirst().
+                    get();
+
+            LOGGER.debug("FinalDestination is {}", finalDestination);
+
+            return finalDestination;
+
+        } catch (Exception e) {
+            LOGGER.warn("Failed in findForEventProcessor: {}", e.getMessage());
+            return empty();
+        }
+
+    }
+
+    private String uriToDestination(final String jmsUri) {
+        try {
+            return jmsUriToDestinationConverter.convert(jmsUri);
+        } catch (Exception e) {
+            return "";
```

Comment:
Is an empty String the best thing to return if the uri conversion fails?

Reply:
This utility is used in stream.filer(something.equalsIgnoreCase(thisMethod(uri))) 
We could return optional... returning empty string here simplifies the code.